src/jit/morph.cpp

```diff
@@ -6560,6 +6560,13 @@ GenTreePtr Compiler::fgMorphField(GenTreePtr tree, MorphAddrContext* mac)
 
             GenTreePtr tlsRef = gtNewIconHandleNode(WIN32_TLS_SLOTS, GTF_ICON_TLS_HDL);
 
+            // Translate GTF_FLD_INITCLASS to GTF_ICON_INITCLASS
```

Comment:
Also line 6468 contains a dangerous SetOper call which can cause the 
flag  GTF_FLD_INITCLASS to be interpreted as GTF_IND_REFARR_LAYOUT